### PR TITLE
feat(react-query): Add user-friendly message when withTRPC is missing from App

### PR DESCRIPTION
**Pull request by kamilogorek**
Closes https://github.com/trpc/trpc/issues/3957

## 🎯 Changes

Prints better error message with the main `_app.tsx` is missing `withTRPC` wrapper.
I'm open to wording suggestions.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/react-query/src/shared/hooks/createHooksInternal.tsx [hunk 1]
```diff
@@ -260,8 +260,14 @@ export function createRootHooks<
     pathAndInput: [path: string, ...args: unknown[]],
     opts?: UseTRPCQueryOptions<unknown, unknown, unknown, unknown, TError>,
   ): UseTRPCQueryResult<unknown, TError> {
+    const context = useContext();
+    if (!context) {
+      throw new Error(
+        'Unable to retrieve application context. Did you forget to wrap your App inside `withTRPC` HoC?',
```
Comment (c-ehrlich) [inline packages/react-query/src/shared/hooks/createHooksInternal.tsx:266]: Wouldn't this also throw in a non-Next.js React app where the provider is missing? Maybe the message can be more general about missing the necessary context/provider?

Comment (kamilogorek) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:266]: Thats exactly what it will do, thus my question about better wording 😅

Comment (bautistaaa) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:266]: could it be better to link to an faq question that has possible answers of why it’s happening(withtrpc missing, provider missing, etc)

**Review by KATT — APPROVED** [review]

File: packages/react-query/src/shared/hooks/createHooksInternal.tsx (hunk 1 above)
Comment (KATT) [reply packages/react-query/src/shared/hooks/createHooksInternal.tsx:277]: Feel free to improve it in a PR, I'm merging this for now!

**Pull request merged**